M2/Macaulay2/packages/Hadamard.m2

```diff
@@ -0,0 +1,441 @@
+newPackage(
+    "Hadamard",
+    Version => "1.0",
+    Date => "November 2020",
+    Authors => {
+
+	{Name => "Iman Bahmani Jafarloo",
+	    Email => "ibahmani89@gmail.com",
+	    HomePage => "http://calvino.polito.it/~imanbj"
+	    }
+	}, -- TODO
+    Headline => "A package for the Hadamard products of varieties",
+    AuxiliaryFiles => false,
+    DebuggingMode => false,
+    Reload => false,
+    PackageExports => {"Points"},
+    Keywords => {"Commutative Algebra"}
+    )
+export {
+    -- types
+    "Point",
+    -- methods
+    "point",
+    "hadamardProduct",
+    "hadamardPower",
+    "idealOfProjectivePoints"
+    }
+
+--defining a new type of objects: points
+
+Point = new Type of BasicList
+
+point=method()
+point(VisibleList):=(P)->(
+    if all(P, i->i==0) then return "error: all entries are zero" else
+    if #delete(null, P) =!= #P then return "error: nuul entries are not allowed"
+    else new Point from P)
+
+makeListOfPoints=method()
+makeListOfPoints(VisibleList):=(L)->(
```

Comment:
This function is superfluous, for in lieu of it, the user can write `point\x`, which is much shorter.

Reply:
I simply deleted it. Since it was used only once and I replaced it with point\x. thanks!